models/src/controller.rs

```diff
@@ -131,10 +135,32 @@ pub fn controller_deployment(
     max_concurrent_update: String,
     update_window_start: String,
     update_window_stop: String,
+    scheduler_cron_expression: String,
 ) -> Deployment {
     let image_pull_secrets =
         image_pull_secret.map(|secret| vec![LocalObjectReference { name: Some(secret) }]);
 
+    // users can specify update time window or scheduler to schedule brupop jobs, but users aren't only allowed to use them simultaneously.
+    let mut scheduler = String::new();
+    if scheduler_cron_expression != SCHEDULER_DEFAULT {
+        if update_window_start != UPDATE_WINDOW_START_DEFAULT.to_string()
+            || update_window_stop != UPDATE_WINDOW_STOP_DEFAULT.to_string()
+        {
+            panic!("You can't use update time window and scheduler simultaneously. Note: the setting update time window is deprecated and you should use the Scheduler setting")
```

Comment:
Is this called by the controller? If so, do you really want to crash the controller and have it start up again?

Reply:
Yeah. I want it to be crashed. This function actually is called when generate yaml file rather than controller container, so we should crash the yaml generation if users config update time window or scheduler in wrong way.

Reply:
To chime in here, even if you want to crash ASAP you do this in Rust by returning an `Err` variant of a `Result` and bubbling it up. Even in this case, I think a `Result` is best-practice but the stakes are low since it's a build-time tool.

Panic is pretty rarely what you want to do.